app/routes.js

```diff
@@ -372,16 +382,18 @@ router.get('/preview-1/:datasetname/:datafileid', function (req, res) {
             var str="";
             response.on('data', data => {
               str += data;
-              // If we've got more than 32000 bytes
-              if (str.length>32000) {
+              var contentLength = response.headers['content-length'] || -1
```

Comment:
I'd guess that response.headers['content-length'] is a string, so you may need to cast it to an integer on the next line ....